recipes/LibriSpeech/ASR/transformer/train_with_whisper.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+""" Take the encoder-decoder Whisper and the official tokenizer of whisper to fine-tune it with the NLL.
+
+This recipe can also be used as a zero-shot recipe by removing the training part.
+
+To run this recipe, do the following:
+> python train_with_whisper.py hparams/train_hf_whisper.yaml
+
+Authors
+ * Adel Moumen 2022
+ * Titouan Parcollet 2022
+"""
+
+import os
+import sys
+import torch
+import logging
+import speechbrain as sb
+from speechbrain.utils.distributed import run_on_main
+from speechbrain.utils.data_utils import undo_padding
+from hyperpyyaml import load_hyperpyyaml
+from pathlib import Path
+from transformers.models.whisper.tokenization_whisper import WhisperTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+# Define training procedure
+class ASR(sb.Brain):
+    def compute_forward(self, batch, stage):
+        """Forward computations from the waveform batches to the output probabilities."""
+        batch = batch.to(self.device)
+        wavs, wav_lens = batch.sig
+        bos_tokens, bos_tokens_lens = batch.tokens_bos
+
+        # Add augmentation if specified
+        if stage == sb.Stage.TRAIN:
+            if hasattr(self.hparams, "augmentation"):
+                wavs = self.hparams.augmentation(wavs, wav_lens)
+
+        abs_tokens_lens = (bos_tokens_lens * bos_tokens.shape[1]).long()
```

Comment:
Add comments to explicit what is happening in the next lines

Reply:
Done.